### PR TITLE
Shortcuts modal

### PR DESCRIPTION
**Pull request by samuel-peers**
Associated Issue: #4122 

### Summary of Changes

Button to open shortcuts modal
Feature flag to disable modal opening

shortcuts_screenshot

### CODE REVIEW

File: src/components/ShortcutsModal.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    return <Modal in={enabled} handleClose={this.closeModal} />;
+  }
+}
+
+ShortcutsModal.propTypes = {
+  enabled: PropTypes.bool
+};
+
+ShortcutsModal.displayName = "ShortcutsModal";
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: we don't need displayName any more!

File: src/components/ShortcutsModal.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      return null;
+    }
+
+    return <Modal in={enabled} handleClose={this.closeModal} />;
+  }
+}
+
+ShortcutsModal.propTypes = {
+  enabled: PropTypes.bool
+};
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: lets switch to using `props`

File: src/components/ShortcutsModal.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// @flow
+import React, { Component, PropTypes } from "react";
+import Modal from "./shared/Modal";
+
+import "./ShortcutsModal.css";
+
+export class ShortcutsModal extends Component {
+  state: SymbolModalState;
+
+  props: {};
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: lets add the flow prop type here:

```js
enabled: boolean
```

File: src/components/SecondaryPanes/UtilsBar.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  };
+
+  return (
+    <button className={classnames(type, className)} {...props}>
+      <Svg name={type} />
+    </button>
+  );
+}
+
+debugBtn.displayName = "UtilsBarButton";
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/UtilsBar.js]: we don't need this

File: src/components/SecondaryPanes/UtilsBar.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        })}
+      >
+        {this.renderUtilButtons()}
+      </div>
+    );
+  }
+}
+
+UtilsBar.propTypes = {
+  toggleShortcutsModal: PropTypes.func
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/UtilsBar.js]: lets remove this

File: src/components/SecondaryPanes/UtilsBar.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    </button>
+  );
+}
+
+debugBtn.displayName = "UtilsBarButton";
+
+class UtilsBar extends Component {
+  props: {
+    horizontal: boolean,
+    toggleShortcutsModal: Function
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/UtilsBar.js]: Lets replace Function w/ `() => void`

File: src/components/SecondaryPanes/UtilsBar.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        "shortcut",
+        "active",
+        "shortcuts",
+        false
+      )
+    ];
+  }
+
+  render() {
+    return (
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/UtilsBar.js]: lets add a feature flag here too

```diff
if(!features.shortcuts) {
  return
}
````

File: src/components/ShortcutsModal.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  render() {
+    const { enabled } = this.props;
+
+    if (!enabled) {
+      return null;
+    }
+
+    return <Modal in={enabled} handleClose={() => console.log("hisam")} />;
```
Comment (jasonLaster) [inline src/components/ShortcutsModal.js]: :) perhaps we should pass `toggleShortcuts` in

File: src/components/App.js [hunk 9]
```diff
@@ -192,7 +197,13 @@ class App extends Component {
             onResizeEnd={size => this.setState({ endPanelSize: size })}
             endPanelControl={true}
             startPanel={this.renderEditorPane()}
-            endPanel={<SecondaryPanes horizontal={horizontal} />}
+            endPanel={
+              <SecondaryPanes
+                horizontal={horizontal}
+                toggleShortcutsModal={() =>
+                  this.setState({ shortcutsModalEnabled: true })}
```
Comment (jasonLaster) [inline src/components/App.js]: hmm, if we make this a true toggle we can re-use it in close modal

```js
this.setState({ shortcutsModalEnabled: !this.state.shortcutsModalEnabled })}
```

File: src/components/ShortcutsModal.js (hunk 8 above)
Comment (samuel-peers) [reply src/components/ShortcutsModal.js]: Hah! that's awesome

**Pull request merged**